login.go

```diff
@@ -0,0 +1,232 @@
+		&cli.StringFlag{
+			Name:        "username",
+			Aliases:     []string{"u"},
+			Usage:       "Username",
+			Destination: &options.username,
+		},
+		&cli.StringFlag{
+			Name:        "password",
+			Aliases:     []string{"p"},
+			Usage:       "Password",
+			Destination: &options.password,
+		},
+		&cli.BoolFlag{
+			Name:        "password-stdin",
+			Usage:       "Take the password from stdin",
+			Destination: &options.passwordStdin,
+		},
+	},
+}
+
+//copied from github.com/docker/cli/cli/command/registry/login.go
+func verifyloginOptions(clicontext *cli.Context, options *loginOptions) error {
+	if options.password != "" {
+		logrus.Warn("WARNING! Using --password via the CLI is insecure. Use --password-stdin.")
+		if options.passwordStdin {
+			return errors.New("--password and --password-stdin are mutually exclusive")
+		}
+	}
+
+	if options.passwordStdin {
+		if options.username == "" {
+			return errors.New("Must provide --username with --password-stdin")
+		}
+
+		contents, err := ioutil.ReadAll(clicontext.App.Reader)
+		if err != nil {
+			return err
+		}
+
+		options.password = strings.TrimSuffix(string(contents), "\n")
+		options.password = strings.TrimSuffix(options.password, "\r")
+	}
+	return nil
+
+}
+
+// Code from github.com/cli/cli/command/registry.go
+// GetDefaultAuthConfig gets the default auth config given a serverAddress
+// If credentials for given serverAddress exists in the credential store, the configuration will be populated with values in it
+func GetDefaultAuthConfig(clicontext *cli.Context, checkCredStore bool, serverAddress string, isDefaultRegistry bool) (*types.AuthConfig, error) {
+	var authconfig = dockercliconfigtypes.AuthConfig{}
+	if checkCredStore {
+		dockerConfigFile, err := dockercliconfig.Load("")
+		if err != nil {
+			return nil, err
+		}
+		authconfig, err = dockerConfigFile.GetAuthConfig(serverAddress)
+		if err != nil {
+			return nil, err
+		}
+	}
+	authconfig.ServerAddress = serverAddress
+	authconfig.IdentityToken = ""
+	res := types.AuthConfig(authconfig)
+	return &res, nil
+}
+
+// Code from github.com/cli/cli/command/registry/login.go
+func loginClientSide(ctx context.Context, auth types.AuthConfig) (registrytypes.AuthenticateOKBody, error) {
+	svc, err := registry.NewService(registry.ServiceOptions{})
+	if err != nil {
+		return registrytypes.AuthenticateOKBody{}, err
+	}
+
+	status, token, err := svc.Auth(ctx, &auth, command.UserAgent())
+
+	return registrytypes.AuthenticateOKBody{
+		Status:        status,
+		IdentityToken: token,
+	}, err
+}
+
+func ConfigureAuthentification(authConfig *types.AuthConfig, options *loginOptions) error {
+	authConfig.Username = strings.TrimSpace(authConfig.Username)
+	if options.username = strings.TrimSpace(options.username); options.username == "" {
+		options.username = authConfig.Username
+	}
+
+	if options.username == "" {
+		return errors.Errorf("error: Username is Required")
+	}
+
+	if options.password == "" {
+		prompt := promptui.Prompt{
```

Comment:
This could be used without violating LGPL https://pkg.go.dev/golang.org/x/crypto@v0.0.0-20201221181555-eec23a3978ad/ssh/terminal

Reply:
(Using any library is fine as long as it works without legal concern)

Reply:
hi @AkihiroSuda  I switched to this lib.